pkg/resourcemonitor/noderesourcesaggregator.go

```diff
@@ -0,0 +1,243 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourcemonitor
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	v1alpha1 "github.com/swatisehgal/topologyapi/pkg/apis/topology/v1alpha1"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	podresourcesapi "k8s.io/kubernetes/pkg/kubelet/apis/podresources/v1alpha1"
+
+	"sigs.k8s.io/node-feature-discovery/pkg/topologyinfo/cpus"
+	"sigs.k8s.io/node-feature-discovery/pkg/topologyinfo/numa"
+	"sigs.k8s.io/node-feature-discovery/pkg/topologyinfo/numa/distances"
+)
+
+const (
+	defaultPodResourcesTimeout = 10 * time.Second
+	// obtained these values from node e2e tests : https://github.com/kubernetes/kubernetes/blob/82baa26905c94398a0d19e1b1ecf54eb8acb6029/test/e2e_node/util.go#L70
+)
+
+type nodeResources struct {
+	perNUMACapacity map[int]map[v1.ResourceName]int64
+	// we should use a different map for each resource type: resourceName -> resourceID -> nodeID
+	resourceID2NUMAID map[string]map[string]int
+	cpuInfo           *cpus.CPUs
+	numaInfo          *numa.Nodes
+	numaDists         *distances.Distances
+}
+
+type ResourceData struct {
+	allocatable int64
+	capacity    int64
+}
+
+func NewResourcesAggregator(sysfsPath string, podResourceClient podresourcesapi.PodResourcesListerClient) (ResourcesAggregator, error) {
+	var err error
+
+	cpuInfo, err := cpus.NewCPUs(sysfsPath)
+	if err != nil {
+		return nil, err
+	}
+
+	nodes, err := numa.NewNodesFromSysFS(sysfsPath)
+	if err != nil {
+		return nil, err
+	}
+
+	dists, err := distances.NewDistancesFromSysfs(sysfsPath)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultPodResourcesTimeout)
+	defer cancel()
+
+	//Pod Resource API client
+	resp, err := podResourceClient.GetAvailableResources(ctx, &podresourcesapi.AvailableResourcesRequest{})
+	if err != nil {
+		return nil, fmt.Errorf("Can't receive response: %v.Get(_) = _, %v", podResourceClient, err)
+	}
+
+	allDevs := GetContainerDevicesFromAllocatableResources(resp, cpuInfo)
+
+	nodeResourceInstance := nodeResources{
+		cpuInfo:           cpuInfo,
+		numaInfo:          &nodes,
+		numaDists:         dists,
+		resourceID2NUMAID: makeResourceMap(len(nodes.Online), allDevs),
+		perNUMACapacity:   MakeNodeCapacity(allDevs),
+	}
+
+	return &nodeResourceInstance, nil
+}
+
```

Comment:
Lets add a comment providing a brief description of what this function does

Reply:
done